### DISCUSSION TITLE
Documentation on VSCode Setup

### DISCUSSION THREAD

**Discussion by vhscom** [micronaut-projects/micronaut-core#8056] [Q&A] [answered]
Hi, I'm new to Micronaut and I work in VSCode. Is there any documentation or example projects (including debug config) for setting up a vanilla Micronaut project with support for step debugging for VSCode? I saw the following discussion but it seems to be more project-specific than what I'm looking for: https://github.com/micronaut-projects/micronaut-core/discussions/7474

**Comment by shingjo** [comment]
There is the information in the VS Code Extension page: https://marketplace.visualstudio.com/items?itemName=oracle-labs-graalvm.micronaut

Is that what you are looking for?

**Reply by vhscom** [reply]
Thanks, @shingjo. VSCode has a debugger built in. Is it not possible to add a debug configuration for that and begin debugging without an extension? Or is the GraalVM extension you linked to required for debugging Micronaut in VSCode?

Aside: Eventually I'd like to be able to start the remote debugger via CLI without any IDE at all.

**Comment by vhscom** [comment, marked as answer]
I was able to get step debugging and inspection working with Micronaut using VSCode without the GraalVM extension. To do so I ran `./gradlew run --debug-jvm` and, from within VS Code, attached a debugger to the port provided on the command line.

<img width="429" alt="Screen Shot 2022-09-20 at 2 32 33 PM" src="https://user-images.githubusercontent.com/97140109/191336953-2ad85a85-b21f-4b90-880e-75358f2fe30a.png">
